lib/modules/manager/custom/jsonata/readme.md

```diff
@@ -0,0 +1,182 @@
+
+#### Required fields to be present in the resulting structure returned by the jsonata query
+
+You must:
+
+- Capture the `currentValue` of the dependency
+- Capture the `depName` or `packageName`. Or use a template field: `depNameTemplate` and `packageNameTemplate`
+- Capture the `datasource`, or a use `datasourceTemplate` config field
+
+#### Optional fields you can include in the resulting structure
+
+You may use any of these items:
+
+- `depType`, or a use `depTypeTemplate` config field
+- `versioning`, or a use `versioningTemplate` config field. If neither are present, Renovate defaults to `semver-coerced`
+- `extractVersion`, or use an `extractVersionTemplate` config field
+- `currentDigest`
+- `registryUrl`, or a use `registryUrlTemplate` config field. If it's a valid URL, it will be converted to the `registryUrls` field as a single-length array
+- `indentation`. It must be either empty, or whitespace only (otherwise `indentation` will be reset to an empty string)
+
+### Usage
+
+To configure it, use the following syntax:
+
+```javascript
+{
+  "customManagers": [
+    {
+      "customType": "jsonata",
+      "fileFormat": "json",
+      "fileMatch": ["<file match pattern>"],
+      "matchStrings": ['<query>'],
+      ...
+    }
+  ]
+}
+```
+
+Where `<query>` is a [JSONata](https://docs.jsonata.org/overview.html) query that transform the contents into a JSON object with the following schema:
+
+To be effective with the JSONata manager, you should understand jsonata queries. But enough examples may compensate for lack of experience.
+
+#### Example queries
+
+Below are some example queries for the generic JSON manager.
+You can also use the [JSONata test website](https://try.jsonata.org) to experiment with queries.
+
+```json title="Dependencies spread in different nodes, and we want to limit the extraction to a particular node"
+{
+  "production": [
+    {
+      "version": "1.2.3",
+      "package": "foo"
+    }
+  ],
+  "development": [
+    {
+      "version": "4.5.6",
+      "package": "bar"
+    }
+  ]
+}
+```
+
+Query:
+
+```
+production.{ "depName": package, "currentValue": version }
+```
+
+```json title="Dependencies spread in different nodes, and we want to extract all of them as if they were in the same node"
+{
+  "production": [
+    {
+      "version": "1.2.3",
+      "package": "foo"
+    }
+  ],
+  "development": [
+    {
+      "version": "4.5.6",
+      "package": "bar"
+    }
+  ]
+}
+```
+
+Query:
+
+```
+*.{ "depName": package, "currentValue": version }
+```
+
+```json title="The dependency name is in a JSON node name and the version is in a child leaf to that node"
+{
+  "foo": {
+    "version": "1.2.3"
+  },
+  "bar": {
+    "version": "4.5.6"
+  }
+}
+```
+
+Query:
+
+```
+$each(function($v, $n) { { "depName": $n, "currentValue": $v.version } })
+```
+
+```json title="The name of the dependency and the version are both value nodes of the same parent node"
```

Comment:
```suggestion
```json title="The dependency name and its version are both value nodes of the same parent node"
```